gomint-api/src/main/java/io/gomint/event/world/SignChangeTextEvent.java

```diff
@@ -48,39 +61,58 @@ public Block getBlock() {
      *
      * @param line    which should be set
      * @param content which should be set on that line
+     * @deprecated Use {@link #setLines(List)} instead
      */
-    public void setLine( int line, String content ) {
+    @Deprecated
+    public void setLine(int line, String content) {
         // Silently fail when line is incorrect
-        if ( line > 4 || line < 1 ) {
+        if (line > 4 || line < 1) {
             return;
         }
 
-        if ( this.lines.size() < line ) {
-            for ( int i = 0; i < line - this.lines.size(); i++ ) {
-                this.lines.add( "" );
+        if (this.lines.size() < line) {
+            for (int i = 0; i < line - this.lines.size(); i++) {
+                this.lines.add("");
             }
         }
 
-        this.lines.set( line - 1, content );
+        this.lines.set(line - 1, content);
     }
 
     /**
      * Get a specific line of the sign content
      *
      * @param line which you want to get
-     * @return string or null when not setö
+     * @return string or null when not set
      */
-    public String getLine( int line ) {
+    public String getLine(int line) {
         // Silently fail when line is incorrect
-        if ( line > 4 || line < 1 ) {
+        if (line > 4 || line < 1) {
             return null;
         }
 
-        if ( this.lines.size() < line ) {
+        if (this.lines.size() < line) {
             return null;
         }
 
-        return this.lines.get( line - 1 );
+        return this.lines.get(line - 1);
+    }
+
+    /**
+     * Get all lines which the sign previously had
+     *
+     * @return all lines which the sign previously had
+     */
+    public List<String> getLines() {
+        return lines;
```

Comment:
Direct manipulation of the amount of sign lines should not be supported. Thus returning the direct list is considered unsafe and should not be possible.

Reply:
Returning list here is not unsafe at all. The sign lines given to the event are a copy of the internal sign lines and they are set into the non-copied array after the event call (with a maximum of 4)

Reply:
It is unsafe as plugins can add or remove lines, possibl causing issues in other plugins whoexpect the lists length to be 4.

Reply:
https://github.com/gomint/gomint/pull/608#discussion_r491692878

Reply:
Resolved by 557b821f1e2c126ac96c0f3757f891d6f15dc512